chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1332,14 +1350,20 @@ impl EthereumAdapterTrait for EthereumAdapter {
             })));
         }
         let hashes: Vec<_> = block.transactions.iter().map(|txn| txn.hash).collect();
-        let receipts_future = fetch_receipts_with_retry(
-            web3,
-            hashes,
+
+        let supports_block_receipts_future = self.check_block_receipt_support_and_update_cache(
```

Comment:
There’s probably a bit more optimization that we can do by reusing the receipt from the check for the actual receipt call, but that would require a bit more refactor and didn’t feel worth it